torch/testing/_internal/common_cuda.py

```diff
@@ -49,6 +50,31 @@ def tf32_is_not_fp32():
     return True
 
 
+@contextlib.contextmanager
+def tf32_off():
+    old_allow_tf32_matmul = torch.backends.cuda.matmul.allow_tf32
+    try:
+        torch.backends.cuda.matmul.allow_tf32 = False
+        with torch.backends.cudnn.flags(enabled=None, benchmark=None, deterministic=None, allow_tf32=False):
```

Comment:
Why is this done with a context manager for cudnn instead of setting the global flag like for matmuls?

Reply:
Because our testing infrastructure do now allow directly setting cudnn flags inside a test